### PR TITLE
[MM-12145] Make user info of user list on two rows: first with username and second with teammate display

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Make user info of user list on two rows: first with username and second with teammate display
- show ellipsis on text overflow of either username or teammate display
- if username is equal to teammate display, only show in one row with username

Note that Add/Manage Members screen is also affected by this change.

#### Ticket Link
Jira ticket: [MM-12145](https://mattermost.atlassian.net/browse/MM-12145)

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Has UI changes

#### Device Information
This PR was tested on: [iOS simulator and Android emulator] 

#### Screenshots
iOS:
screen shot 2018-09-28 at 3 20 56 am

screen shot 2018-09-28 at 3 21 06 am

screen shot 2018-09-28 at 3 21 51 am

screen shot 2018-09-28 at 3 21 59 am

Android:
screen shot 2018-09-28 at 3 36 18 am

screen shot 2018-09-28 at 3 36 42 am

Other screen affected by the change on CustomListRow which looks the same as before: 

Add Members Screen:
screen shot 2018-09-28 at 3 38 15 am

Manage Members screen:
screen shot 2018-09-28 at 3 47 40 am

Mor Channels screen:
screen shot 2018-09-28 at 4 28 47 pm

### CODE REVIEW

File: app/components/custom_list/custom_list_row.js [hunk 1]
```diff
@@ -49,7 +49,9 @@ export default class CustomListRow extends React.PureComponent {
                             </View>
                         </View>
                     }
-                    {this.props.children}
+                    <View style={style.children}>
```
Comment (enahum) [inline app/components/custom_list/custom_list_row.js]: what other component or list is being affected by this?

File: app/components/custom_list/user_list_row/user_list_row.js [hunk 2]
```diff
@@ -63,87 +65,83 @@ export default class UserListRow extends React.PureComponent {
...
+                                numberOfLines={1}
+                            >
+                                {teammateDisplay}
+                            </Text>
+                        </View>
+                        }
                     </View>
-                </View>
-            </CustomListRow>
+                    <View style={{width: 25}}/>
```
Comment (saturninoabril) [inline app/components/custom_list/user_list_row/user_list_row.js]: Above is just a filler on right side.  Don't know why adding margin or padding to the right/horizontal does not work for me.

File: app/components/custom_list/user_list_row/user_list_row.js [hunk 3]
```diff
@@ -63,87 +65,83 @@ export default class UserListRow extends React.PureComponent {
...
-            alignItems: 'center',
+        hideTeammateDisplay: {
             justifyContent: 'center',
         },
-        selectorDisabled: {
-            backgroundColor: '#888',
-        },
-        selectorFilled: {
-            backgroundColor: '#378FD2',
-            borderWidth: 0,
```
Comment (saturninoabril) [inline app/components/custom_list/user_list_row/user_list_row.js]: These selector styles are not used by this component.

File: app/components/custom_list/custom_list_row.js [hunk 4]
```diff
@@ -65,9 +67,8 @@ const getStyleFromTheme = makeStyleSheetFromTheme((theme) => {
             alignItems: 'center',
             backgroundColor: theme.centerChannelBg,
         },
-        displayName: {
```
Comment (saturninoabril) [inline app/components/custom_list/custom_list_row.js]: displayName is not used by this component.

File: app/components/custom_list/user_list_row/user_list_row.js (hunk 2 above)
Comment (enahum) [reply app/components/custom_list/user_list_row/user_list_row.js]: ok but make it a style

File: app/components/custom_list/custom_list_row.js (hunk 1 above)
Comment (saturninoabril) [reply app/components/custom_list/custom_list_row.js]: Affected components/screens are the following and looks the same as before after this change.  See screenshots of screens on description.

CustomListRow
- ChannelListRow
    - CustomFlatList
        - MoreDirectMessages (screen)
    - CustomList
        - ChannelAddMembers (screen)
    - CustomSectionList
        - MoreDirectMessages (screen)
    - MenuActionSelector (screen)
    - MoreChannels (screen)
- OptionListRow
    - MenuActionSelector (screen)
- UserListRow
    - CustomFlatList (see above)
    - CustomSectionList (see above)
    - ChannelAddMembers (screen)
    - ChannelMembers (screen)
    - MenuActionSelector (screen)
    - MoreDirectMessages (screen)

**Comment by saturninoabril** [comment]
@enahum @hmhealey This is now ready for review.

**Comment by hmhealey** [comment]
Code looks good to me

@esethna How do you feel about the alignment of the text vs the profile pictures for the user lists with this? It looks like it's aligned with the checkbox and and the combined profile picture/status icon, but that makes it look a bit off since it's not aligned with the profile picture.

image

That being said, I doubt this is a new issue. We've had this problem in other places because of how the status icon is offset

**Comment by enahum** [comment]
@esethna  if needed lets create a ticket for what @hmhealey pointed out, for now I'll merge this

**Pull request merged**